source/common/http/http1/codec_impl.cc

```diff
@@ -626,9 +622,9 @@ void ConnectionImpl::onMessageBeginBase() {
   // protocol for each request. Envoy defaults to 1.1 but sets the protocol to 1.0 where applicable
   // in onHeadersCompleteBase
   protocol_ = Protocol::Http11;
-  ASSERT(!current_header_map_);
```

Comment:
think it's worth keeping this assert in maybeAlloc...?
I like the guarantees we had before that we never leaked headers/trailers across sequential streams, at least until we move this to ActiveRequest/PendingResponse per your TODO